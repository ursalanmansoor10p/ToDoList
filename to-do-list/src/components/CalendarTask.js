import React from "react";

function CalendarTask({ todos }) {
  //console.log("calendar",todos)
  //A commented out console.log to see props object

  //Filter through the todos prop. The prop holds object
  //db prop that was passed {todos}
  const filteredTodos = todos.filter((todos) => todos.completed === true);

  // filtered through db console.logged out
  console.log("ok", filteredTodos);

  return (
    <div className="to" data-testid="calendar-tasks">
      {!(filteredTodos.length > 0) ? (
        <p>Sorry there are no completed tasks</p>
      ) : (
        <ul>
          {/* .map goes through all the arrats to print each db.title  */}
          {filteredTodos.map((task) => (
            <li key={task.id} style={{ marginBottom: "20px" }}>
              {task.title}
            </li>
          ))}
        </ul>
      )}
    </div>
  );
}

export default CalendarTask;
