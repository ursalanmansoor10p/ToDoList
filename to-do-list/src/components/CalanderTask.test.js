import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import CalendarTask from "./CalendarTask";

const todos = [
  { Title: "ateta", completed: false },
  { Title: "ateeeta", completed: true },
];

describe("CalendarTask Component", () => {
  test("should match snapshot without todos", () => {
    const { container } = render(<CalendarTask todos={[]} />);
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot with todos", () => {
    const { container } = render(<CalendarTask todos={todos} />);
    expect(container).toMatchSnapshot();
  });

  test("should list completed tasks", () => {
    render(<CalendarTask todos={todos} />);
    const li = screen.getAllByRole("listitem");
    expect(li.length).toBe(1);
  });

  test("should display a message if no task is completed", () => {
    render(<CalendarTask todos={[{ Title: "abc", completed: false }]} />);
    const li = screen.queryAllByRole("listitem");
    expect(li.length).toBe(0);
    const p = screen.getByText("Sorry there are no completed tasks");
    expect(p).toBeInTheDocument();
  });
});
