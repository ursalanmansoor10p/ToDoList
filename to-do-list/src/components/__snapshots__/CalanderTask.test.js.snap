// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarTask Component should match snapshot with todos 1`] = `
<div>
  <div
    class="to"
    data-testid="calendar-tasks"
  >
    <ul>
      <li
        style="margin-bottom: 20px;"
      />
    </ul>
  </div>
</div>
`;

exports[`CalendarTask Component should match snapshot without todos 1`] = `
<div>
  <div
    class="to"
    data-testid="calendar-tasks"
  >
    <p>
      Sorry there are no completed tasks
    </p>
  </div>
</div>
`;
